sim/paladin/item_sets_pve.go

```diff
@@ -0,0 +1,136 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/classic/sim/core"
+	"github.com/wowsims/classic/sim/core/proto"
+	"github.com/wowsims/classic/sim/core/stats"
+)
+
+///////////////////////////////////////////////////////////////////////////
+//                            Classic Phase 1 Item Sets - Molten Core
+///////////////////////////////////////////////////////////////////////////
+
+var ItemSetVestmentsOfProphecy = core.NewItemSet(core.ItemSet{
+	Name: "Lawbringer Armor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Increases the chance of triggering a Judgement of Light heal by 10%.
+		3: func(agent core.Agent) {
+			// Nothing to do
+		},
+		// Improves your chance to get a critical strike with spells by 1%.
+		// Improves your chance to get a critical strike by 1%.
+		5: func(agent core.Agent) {
+			paladin := agent.(PaladinAgent).GetPaladin()
+			paladin.AddStat(stats.MeleeCrit, 1)
+			paladin.AddStat(stats.SpellCrit, 1)
+		},
+		// Gives the Paladin a chance on every melee hit to heal your party for 189 to 211.
+		8: func(agent core.Agent) {
+			// Nothing to do
+		},
+	},
+})
+
+///////////////////////////////////////////////////////////////////////////
+//                            Classic Phase 3 Item Sets - BWL
+///////////////////////////////////////////////////////////////////////////
+
+var ItemSetSoulforgeArmor = core.NewItemSet(core.ItemSet{
+	Name: "Judgement Armor",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Increases the radius of a Paladin's auras by 10.
+		3: func(agent core.Agent) {
+			// Nothing to do
+		},
+		// Increases damage and healing done by magical spells and effects by up to 47.
+		5: func(agent core.Agent) {
+			c := agent.GetCharacter()
+			c.AddStat(stats.SpellPower, 47)
+		},
+		// Inflicts 60 to 66 additional Holy damage on the target of a Paladin's Judgement.
+		8: func(agent core.Agent) {
+			paladin := agent.(PaladinAgent).GetPaladin()
+			paladin.RegisterAura(core.Aura{
+				Label: "Judgement - T2 - Paladin - 8P Bonus",
+				SpellSchool: core.SpellSchoolHoly,
+				OnInit: func(aura *core.Aura, sim *core.Simulation) {
+					for _, judgeSpells := range paladin.allJudgeSpells {
+						for _, judgeRankSpell := range judgeSpells {
+							spell.CalcAndDealDamage(sim, target, sim.Roll(60, 66), spell.OutcomeMagicCrit)
+						}
+					}
+				},
```

Comment:
🛑 this won't quite work correctly. What's happening here is when the sim first initializes you're effectively dealing damage with each judgement, not when the judgement is actually cast. I think what you want to do is use `core.MakePermanent` to make this aura a permanent aura, then add an `OnSpellHitDealt` handler that checks if the spell is a judgement and if the spell landed, and if that's true then deal the extra damage